File: Deafios em C#/1040 - Média 3.cs
using System; 
using System.Globalization;

//Developed by: @LucasMarcuzo
class URI1040 {

 static void Main(string[] args){ 

	float n1, n2, n3, n4, media, exame;

    string[] s = Console.ReadLine().Split(' ');

    n1 = float.Parse(s[0], CultureInfo.InvariantCulture);
    n2 = float.Parse(s[1], CultureInfo.InvariantCulture);
    n3 = float.Parse(s[2], CultureInfo.InvariantCulture);
    n4 = float.Parse(s[3], CultureInfo.InvariantCulture);

	media = ((n1*2 + n2*3 + n3*4 + n4*1) / (2 + 3 + 4 + 1));


	if (media >= 7.0)
    {

        Console.WriteLine("Media: " + media.ToString("F1", CultureInfo.InvariantCulture));
		Console.WriteLine("Aluno aprovado.");

	}
	else
    {

		if (media < 5.0)
        {

            Console.WriteLine("Media: {0:f1}", (media-0.01).ToString("F1", CultureInfo.InvariantCulture));

			Console.WriteLine("Aluno reprovado.");

		}
		else
        {

            Console.WriteLine("Media: {0:f1}", (media-0.01).ToString("F1", CultureInfo.InvariantCulture));
			Console.WriteLine("Aluno em exame.");
            exame = float.Parse(Console.ReadLine(), CultureInfo.InvariantCulture);
            Console.WriteLine("Nota do exame: " + exame.ToString("F1", CultureInfo.InvariantCulture));
			media = (media + exame)/2;

			if (media >= 5.0)
            {

				Console.WriteLine("Aluno aprovado.");

			}
			else
            {

				Console.WriteLine("Aluno reprovado.");

			}

            Console.WriteLine("Media final: " + media.ToString("F1", CultureInfo.InvariantCulture));

		    }
        }
	}
}

File: Desafios em JavaScript/1040 - Média 3.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

if(lines.length > 1){
    var ln1 = lines.shift().split(' ');
    var ln2 = lines.shift().split(' ');
}
else var ln1 = lines.shift().split(' ');  

var n1, n2, n3, n4, media, exame;

n1 = parseFloat(ln1.shift());
n2 = parseFloat(ln1.shift());
n3 = parseFloat(ln1.shift());
n4 = parseFloat(ln1.shift());

media = parseFloat(((n1*2 + n2*3 + n3*4 + n4*1) / (2 + 3 + 4 + 1)));

if (media >= 7.0){

    console.log(`Media: ${media.toFixed(1)}`);
    console.log(`Aluno aprovado.`);

}
else{

    if (media < 5.0){

        media -= 0.01;
        console.log(`Media: ${media.toFixed(1)}`);
        console.log(`Aluno reprovado.`);

    }
    else{

        media -= 0.01;
        console.log(`Media: ${media.toFixed(1)}`);
        console.log(`Aluno em exame.`);
        exame = parseFloat(ln2.shift());
        console.log(`Nota do exame: ${exame.toFixed(1)}`);
        media = (media + exame)/2;
        if (media >= 5.0){
            console.log(`Aluno aprovado.`);
        }
        else{
            console.log(`Aluno reprovado.`);
        }

        console.log(`Media final: ${media.toFixed(1)}`);
    }
}